quickwit/quickwit-metastore/src/tests/shard.rs

```diff
@@ -95,12 +104,50 @@ pub async fn test_metastore_open_shards<
     )
     .await;
 
+    // Test empty request.
     let open_shards_request = OpenShardsRequest {
         subrequests: Vec::new(),
     };
     let open_shards_response = metastore.open_shards(open_shards_request).await.unwrap();
     assert!(open_shards_response.subresponses.is_empty());
 
+    // Test index not found.
+    let open_shards_request = OpenShardsRequest {
+        subrequests: vec![OpenShardsSubrequest {
+            index_uid: "index-does-not-exist:0".to_string(),
+            source_id: test_index.source_config.source_id.clone(),
+            leader_id: "test-ingester-foo".to_string(),
+            next_shard_id: 1,
+            ..Default::default()
+        }],
+    };
+    let error = metastore
+        .open_shards(open_shards_request)
+        .await
+        .unwrap_err();
+    assert!(
+        matches!(error, MetastoreError::NotFound(EntityKind::Index { index_id }) if index_id == "index-does-not-exist")
+    );
+
+    // Test source not found.
```

Comment:
@kamalesh0406, I don't think this will be easy to implement for Postgres. Feel free to comment.

Reply:
I feels this looks promising https://docs.rs/sqlx/latest/sqlx/attr.test.html . What are your thoughts?